sno/fast_import.py

```diff
@@ -311,48 +395,54 @@ def _ids():
 
                 # Meta items - written second as certain importers generate extra metadata as they import features.
                 for x in write_blobs_to_stream(
-                    p.stdin, dataset.import_iter_meta_blobs(repo, source)
+                    procs[0].stdin, dataset.import_iter_meta_blobs(repo, source)
                 ):
                     pass
 
-        p.stdin.write(b"\ndone\n")
-    except BrokenPipeError:
-        # if git-fast-import dies early, we get an EPIPE here
-        # we'll deal with it below
-        pass
-    else:
-        p.stdin.close()
-    p.wait()
-    if p.returncode != 0:
-        raise SubprocessError(
-            f"git-fast-import error! {p.returncode}", exit_code=p.returncode
-        )
     t3 = time.monotonic()
     if verbosity >= 1:
         click.echo(f"Closed in {(t3-t2):.0f}s")
 
-    if import_branch is not None:
-        # we created a temp branch for the import above.
+    if import_branches:
```

Comment:
Feels like it's worth adding status/logging around this so we can ascertain how long the various steps take?